src/data/mv_polynomial/expand.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Robert Y. Lewis
+-/
+import data.mv_polynomial.monad
+
+/-!
+## Expand multivariate polynomials
+
+Given a multivariate polynomial `φ`, one may replace every occurence of `X i` by `X i ^ n`,
+for some natural number `n`.
+This operation is called `mv_polynomial.expand` and it is an algebra homomorphism.
+
+### Main declaration
+
+* `mv_polynomial.expand`: expand a polynomial by a factor of p, so `∑ aₙ xⁿ` becomes `∑ aₙ xⁿᵖ`.
+-/
+
+open_locale big_operators
+
+namespace mv_polynomial
+
+variables {σ τ R S : Type*} [comm_semiring R] [comm_semiring S]
+
+/-- Expand the polynomial by a factor of p, so `∑ aₙ xⁿ` becomes `∑ aₙ xⁿᵖ`.
+
+See also `polynomial.expand`. -/
+noncomputable def expand (p : ℕ) : mv_polynomial σ R →ₐ[R] mv_polynomial σ R :=
+{ commutes' := λ r, eval₂_hom_C _ _ _,
+  .. (eval₂_hom C (λ i, (X i) ^ p) : mv_polynomial σ R →+* mv_polynomial σ R) }
+
+@[simp] lemma expand_C (p : ℕ) (r : R) : expand p (C r : mv_polynomial σ R) = C r :=
+eval₂_hom_C _ _ _
+
+@[simp] lemma expand_X (p : ℕ) (i : σ) : expand p (X i : mv_polynomial σ R) = (X i) ^ p :=
+eval₂_hom_X' _ _ _
+
+@[simp] lemma expand_monomial (p : ℕ) (d : σ →₀ ℕ) (r : R) :
+  expand p (monomial d r) = C r * ∏ i in d.support, ((X i) ^ p) ^ (d i) :=
+bind₁_monomial _ _ _
+
+lemma expand_one_apply (f : mv_polynomial σ R) : expand 1 f = f :=
+by simp only [expand, bind₁_X_left, alg_hom.id_apply, ring_hom.to_fun_eq_coe,
+  eval₂_hom_C_left, alg_hom.coe_to_ring_hom, pow_one, alg_hom.coe_mk]
```

Comment:
On my machine, `simp` is fast enough for this goal.
```suggestion
by simp [expand]
```

Reply:
But I fear that your machine is not Kenny's machine.